### PR TITLE
Add ContextMenu, DateInput, EmailLink, ErrorMessage, HorizontalDivider and Input

### PR DESCRIPTION
**Pull request by winston0410**

### CODE REVIEW

File: src/components/ui/ContextMenu/ContextMenu.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      onClick: () => {},
+    },
+    {
+      icon: <GarageIcon />,
+      label: 'This is the garage label',
+      onClick: () => {},
+    },
+  ],
+  icon: <MoreIcon />,
+  testId: '',
```
Comment (vadim-kirpun) [inline src/components/ui/ContextMenu/ContextMenu.stories.tsx:32]: No need to pass `testId` in storybook

File: src/components/ui/ContextMenu/ContextMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  onClick?: () => void;
+};
+
+type ContextMenuProps = {
+  options: Array<ContextMenuOption>;
+  trigger?: ReactNode;
+  testId?: string | null;
+};
+
+//  <MoreIcon />
```
Comment (vadim-kirpun) [inline src/components/ui/ContextMenu/ContextMenu.tsx:29]: Let's remove this comment

File: src/components/ui/DateInput/DateInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  component: DateInput,
+  argTypes: {},
+} as ComponentMeta<typeof DateInput>;
+
+const Template: ComponentStory<typeof DateInput> = (args) => <DateInput {...args} />;
+
+export const Normal = Template.bind({});
+Normal.args = {
+  label: 'Input field label',
+  dateValues: {
```
Comment (vadim-kirpun) [inline src/components/ui/DateInput/DateInput.stories.tsx:15]: Let's share this object between stories

File: src/components/ui/DateInput/DateInput.test.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      day: '30',
+      month: '02',
+      year: '2000',
+    },
+  };
+
+  it('renders with given class', () => {
+    render(<DateInput {...props} className="foo" />);
+
+    const content = screen.getAllByTestId('left-border-container')[0];
```
Comment (vadim-kirpun) [inline src/components/ui/DateInput/DateInput.test.tsx:17]: There should be only one LeftBorderContainer, shouldn't it?

File: src/components/ui/DateInput/DateInput.test.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    expect(content).toBeInTheDocument();
+  });
+
+  it('renders with given error message', () => {
+    render(<DateInput {...props} errorMessage="foo" />);
+
+    const content = screen.getByText('foo');
+
+    expect(content).toBeInTheDocument();
+  });
```
Comment (vadim-kirpun) [inline src/components/ui/DateInput/DateInput.test.tsx:36]: let's test if `onChange` works

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Comment by vadim-kirpun** [comment]
There is no trigger passed for `ContextMenu`
image

File: src/components/ui/ContextMenu/ContextMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+const ContextMenu = ({ options, trigger, testId = null }: ContextMenuProps) => {
```
Comment (vadim-kirpun) [inline src/components/ui/ContextMenu/ContextMenu.tsx:30]: trigger should be `<MoreIcon />` by default

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Comment by vadim-kirpun** [comment]
Let's add one more story for `DateInput` - with error and left border 
So we can see that border is red

File: src/components/ui/ContextMenu/ContextMenu.tsx (hunk 6 above)
Comment (winston0410) [reply src/components/ui/ContextMenu/ContextMenu.tsx:30]: I am setting that as default in line 80. If you really want me to set it as default here, we have to capitalize trigger

https://stackoverflow.com/questions/71018466/how-to-define-a-component-as-default-value-for-a-prop-in-react/71018639#71018639

File: src/components/ui/DateInput/DateInput.test.tsx (hunk 4 above)
Comment (winston0410) [reply src/components/ui/DateInput/DateInput.test.tsx:17]: Not really, I think they are nested here, thats why I have to use it like this

Comment (vadim-kirpun) [reply src/components/ui/DateInput/DateInput.test.tsx:17]: Yeah you're right, it's nested inside of `Input`, ok

File: src/components/ui/ContextMenu/ContextMenu.tsx (hunk 6 above)
Comment (vadim-kirpun) [reply src/components/ui/ContextMenu/ContextMenu.tsx:30]: That's fine
The problem is that I don't see this default icon in storybook if we're not passing trigger

Comment (winston0410) [reply src/components/ui/ContextMenu/ContextMenu.tsx:30]: So do you mean that you are fine with me not setting the default icon there, as long as I have fixed it?

Comment (vadim-kirpun) [reply src/components/ui/ContextMenu/ContextMenu.tsx:30]: Yes. the main think is to see MoreIcon as default in storybook

Comment (winston0410) [reply src/components/ui/ContextMenu/ContextMenu.tsx:30]: I think you have missed the trigger there. The trigger is located at the right on your screen.

**Comment by vadim-kirpun** [comment]
image
The border is not red, let's pass `errorMessage` to paint it
`withError` is used only to color input in red, there is a comment about it in the props list

**Comment by vadim-kirpun** [comment]
image
There is still no `MoreIcon`

**Comment by vadim-kirpun** [comment]
I don't see any effect from ReactDocgenTypescriptPlugin
What should it do?

**Comment by vadim-kirpun** [comment]
There are many TS errors, please check the code with `tsc`. Please investigate why they were pushed

**Comment by vadim-kirpun** [comment]
Some tests are failed

**Comment by winston0410** [comment]
> image There is still no `MoreIcon`

@vadim-kirpun Can you see 3 dot or your right????

File: src/hooks/hooks.test.ts [hunk 7]
```diff
@@ -34,3 +34,20 @@ describe('useTabs', () => {
...
   });
 });
+
+describe('useDateInput', () => {
+  it('returns correct props', () => {
+    // REF https://stackoverflow.com/questions/45644098/testing-anonymous-function-equality-with-jest
+    const setStateMock = expect.any(Function);
+    const { result } = renderHook(() => useDateInput());
+
+    expect(result.current).toEqual({
```
Comment (vadim-kirpun) [inline src/hooks/hooks.test.ts:44]: This check doesn't do anything useful
We need to do something like this `jest.spyOn(React, 'useState').mockImplementation(...)`
And return some specific dateValues object
We need two cases for this, so we can check that:
1. If we specify all day/month/year - isFilled will be true, resultDate is calculated
2. If we specify only day - isFilled will be false, isPartiallyEmpty will be true, resultDate is not ready

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Pull request merged**